### ISSUE TITLE
-P in extra options fails

### ISSUE THREAD

**Issue by Ralt** [geany/geany#443]
Hi,

When adding `-P` to the extra options of "Find in files", this error appears:

> /bin/grep: conflicting matchers specified

This makes it impossible to use negative lookbehind searches, for examples.

**Comment by ntrel** [comment]
Probably because you have 'Use regular expressions' checked, which passes '-E' (you can see this in the first line of Messages window IIRC).

**Comment by Ralt** [comment]
@ntrel yep, it uses either `-F` or `-E`. https://github.com/geany/geany/blob/e9c9e9a2fa021e22ab3235b089c3e651f0094016/src/search.c#L1566-L1569

**Comment by Canop** [comment]
When I uncheck 'Use regular expressions' it still gives me the same error

**Comment by Canop** [comment]
When I uncheck 'Use regular expressions' it still gives me the same error

**Comment by b4n** [comment]
As a workaround, you can use a custom script that will wrap the `grep` call and replace the `-E` option with `-P`:

```shell
#!/bin/sh

matchopts=$(echo "$1" | tr E P)
shift

exec grep $matchopts "$@"
```

and use this script as `Preferences → Tools → Grep`.

**Comment by kugel-** [comment]
Should probably make the argument is used for regexp configurable
